File: backend/app/schemas/issue.py
"""API response schemas.

Schemas define the shape of data returned by the FastAPI backend. They keep
the API response predictable for the Nuxt frontend and for tools like Postman.
"""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class GitHubIssueResponse(BaseModel):
    """Public API representation of a collected GitHub issue."""

    id: int
    github_issue_id: int
    github_issue_number: int

    github_repository_owner: str
    github_repository_name: str
    github_repository_full_name: str

    github_issue_url: str
    github_repository_url: str

    title: str
    body_preview: str | None
    state: str
    labels: list[dict[str, Any]]

    is_assigned: bool
    assignee_count: int
    comment_count: int

    repository_primary_language: str | None
    repository_stars: int | None
    repository_forks: int | None
    repository_open_issues_count: int | None
    repository_is_archived: bool | None
    repository_is_disabled: bool | None

    source_query: str
    collected_by: str
    collected_at: datetime

    github_created_at: datetime
    github_updated_at: datetime
    github_closed_at: datetime | None

    model_config = ConfigDict(from_attributes=True)

File: frontend/app/types/issue.ts
export interface GitHubIssue {
  id: number;
  github_issue_id: number;
  github_issue_number: number;
  github_repository_owner: string;
  github_repository_name: string;
  github_repository_full_name: string;
  github_issue_url: string;
  github_repository_url: string;
  title: string;
  body_preview: string | null;
  state: string;
  labels: Array<Record<string, unknown>>;
  is_assigned: boolean;
  assignee_count: number;
  comment_count: number;
  repository_primary_language: string | null;
  repository_stars: number | null;
  repository_forks: number | null;
  repository_open_issues_count: number | null;
  repository_is_archived: boolean | null;
  repository_is_disabled: boolean | null;
  source_query: string;
  collected_by: string;
  collected_at: string;
  github_created_at: string;
  github_updated_at: string;
  github_closed_at: string | null;
}
